documentation/release_notes.markdown

```diff
@@ -61,179 +94,253 @@ This release retires all api versions before v4 and promotes v4 to be
 the stable API. The following changes have been made since the last
 stable release of PuppetDB, which included an experimental v4 API.
 
-* Retire all previously deprecated commands ([PDB-840](https://tickets.puppetlabs.com/browse/PDB-840))
+* Versions 2 and 3 of the query API are now retired. There are also significant differences between the current stable v4 and the previously experimental v4.
+
+    Full documentation outlining this change is available [in the API upgrading guide][upgrading].
+
+* We've retired some previously deprecated commands ([PDB-840](https://tickets.puppetlabs.com/browse/PDB-840))
 
     The currently supported commands and versions are now:
 
     * replace catalog, v6
     * replace facts, v4
-    * deactivate node, v2
+    * deactivate node, v3
     * store report, v5
 
-* All API commands and responses now use underscores, instead of dashes.  ([PDB-698](https://tickets.puppetlabs.com/browse/PDB-698))
+    Older versions of the commands on your queue will still be
+    processed by PuppetDB so you won't lose the data from messages
+    still being processed by PuppetDB before an upgrade.
+
+* All API commands and responses now use underscores, instead of dashes. ([PDB-698](https://tickets.puppetlabs.com/browse/PDB-698))
 
     This patch changes our API commands and responses to use underscores instead of
     dashes, which will make it easier to interface with PDB using languages like
-    javascript and python.
-
-* Split the `/v3/commands` endpoint into its own API served at `/pdb/cmd/v1`.
+    javascript and python. Consult the various API endpoints for the exact new
+    parameters if you are uncertain.
+
+* Split the command endpoint (`/v3/commands` or `/v4/commands`) into its own
+  endpoint namespace served at `/pdb/cmd/v1` so it can be versioned and managed separately. ([PDB-1534](https://tickets.puppetlabs.com/browse/PDB-1534)) ([PDB-1561](https://tickets.puppetlabs.com/browse/PDB-1561))
+
+* The `/metrics` endpoint now has its own top level namespace `/metrics/v1` ([PDB-800](https://tickets.puppetlabs.com/browse/PDB-800))
+
+    This is something we'd been planning on doing for a while, but separation
+    means we can manage and version this endpoint separately. There are plans
+    to have this endpoint supported in other Puppet Labs products also, and
+    this move gears us up for that.
+
+* Version and server-time are now mounted under the metadata API ([PDB-1562](https://tickets.puppetlabs.com/browse/PDB-1562))
+
+    These endpoints now live in `/pdb/meta/v1`.
```

Comment:
Similarly for `cmd`, I think maybe these three things could all be in their own chunk.